src/urllib3/util/ssl_.py

```diff
@@ -291,6 +291,13 @@ def create_urllib3_context(
 
     context.options |= options
 
+    # Enable post-handshake authentication for TLS 1.3, see GH #1634. PHA is
+    # necessary for conditional client cert authentication with TLS 1.3.
+    # The attribute is None for OpenSSL <= 1.1.0 or does not exist in older
+    # versions of Python.
+    if getattr(context, "post_handshake_auth", None) is not None:
```

Comment:
If `context.post_handshake_auth` is False, this branch (which sets it to True) will still be taken. Would this be better written without the `is not None`? (Even if that’s never expected, it seems like cleaner style here?)

Reply:
Oh, just noticed the (False, True) test case below, so I guess this is intended (and now I’m just confused why, but please feel free to ignore and sorry for the noise).

Reply:
@jab Don't be sorry for the noise! I'm really glad that you took the time to review this pull request. And we should probably take more time trying to understand the code we're merging, here we simply applied the commit from https://github.com/urllib3/urllib3/pull/1635/files and did not look hard at it because it was already reviewed upstream.

But you're right, this is debatable. Maybe the reasoning is that the only values that we really care about are None (default value, disabled) and True (enabled)? I asked the question in the original pull request: https://github.com/urllib3/urllib3/pull/1635/files#r333342481

Reply:
Oh and I forgot to mention it, but https://github.com/python-trio/urllib3/pull/128 is currently ready for review if you'd like to take a look. :)